homeassistant/components/bayesian/binary_sensor.py

```diff
@@ -292,32 +315,31 @@ def _async_template_result_changed(event, updates):
             )
 
     @callback
-    def _recalculate_and_write_state(self):
+    def _recalculate_and_write_state(self) -> None:
         self.probability = self._calculate_new_probability()
         self._attr_is_on = bool(self.probability >= self._probability_threshold)
         self.async_write_ha_state()
 
-    def _initialize_current_observations(self):
-        local_observations = OrderedDict({})
-
+    def _initialize_current_observations(self) -> OrderedDict[str, Observation]:
+        local_observations: OrderedDict[str, Observation] = OrderedDict({})
         for entity in self.observations_by_entity:
             local_observations.update(self._record_entity_observations(entity))
         return local_observations
 
-    def _record_entity_observations(self, entity):
+    def _record_entity_observations(self, entity: str) -> OrderedDict[str, Observation]:
         local_observations = OrderedDict({})
 
         for observation in self.observations_by_entity[entity]:
             platform = observation.platform
 
-            observed = self.observation_handlers[platform](observation)
+            observed: bool | None = self.observation_handlers[platform](observation)
```

Comment:
Not strictly necessary and happy to remove if the reviewer wants
Put in for readability to clarify that it can be None

Reply:
As you said, it is not necessary (`self.observation_handlers` is typed, so this is inferred)
For my part, I would simply drop the observed variable, and assign directly:
```python
observation.observed = self.observation_handlers[platform](observation)
```

Reply:
done